velox/exec/WindowFunction.h

```diff
@@ -31,6 +31,22 @@ struct WindowFunctionArg {
   std::optional<const column_index_t> index;
 };
 
+/// The processing unit for calculating the window function in a streaming
+/// manner. kRows indicates that the calculation begins as soon as rows are
+/// available within a single partition, without waiting for all data in the
+/// partition to be ready. kPartition indicates that the calculation begins only
+/// when all rows in a partition are ready.
+enum class ProcessingUnit {
+  kPartition,
+  kRows,
+};
+
+/// Store the metadata for WindowFunction.
+struct StreamingProcessMetadata {
```

Comment:
Nit : Maybe WindowFunctionMetadata is a more generic name in the long run.

Reply:
@aditi-pandit Yes. Updated.